test/unit/clarity_ext/dilution/hamilton_tests.py

```diff
@@ -17,6 +20,9 @@ class HamiltonTests(unittest.TestCase):
     """
 
     def setUp(self):
+        self.mock = DilutionMock1()
```

Comment:
There are a few things I would change here:
 * Skip that helper class called `FakingEntityMonkey`
 * Move all mock logic from `DilutionMock1` into the test itself - possibly dragging in the mock library. My view of tests is that they should generally be the only thing one needs to read in order to understand a unit.
 * Mock the methods with either monkey patching or by using mock objects from the mock library.

This would get rid of several classes as well as increasing readability (perhaps arguably).
 
I'll see if I can sketch my ideas in another branch real quick and see what you think.